From 9621224dba9fc0fe1efa4884ff2ef5d1f44c7827 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Milan=20P=2E=20Stani=C4=87?= <mps@arvanta.net>
Date: Sat, 9 Aug 2025 14:23:51 +0000
Subject: [PATCH] quick fix to build with gcc 15 because of this error: note:
 'bool' is a keyword with '-std=c23' onwards

simply add 'x' prefix to bool variable
---
 interface.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/interface.c b/interface.c
index 3e92965..6c7f0b5 100644
--- a/interface.c
+++ b/interface.c
@@ -757,16 +757,16 @@ int interface_config(struct configfile *cf)
 #endif
 		} else if (strcmp(name,"tx-ok") == 0) {
 
-                  int bool;
-		  if (!config_parse_boolean(param1, &bool)) {
+                  int xbool;
+		  if (!config_parse_boolean(param1, &xbool)) {
 		    printf("%s:%d ERROR: Bad TX-OK parameter value -- not a recognized boolean: %s\n",
 			   cf->name, cf->linenum, param1);
 		    have_fault = 1;
 		    continue;
 		  }
-                  aif->tx_ok = bool;
-		  if (bool && aif->callsign) {
-		    if (!validate_callsign_input(aif->callsign,bool)) {  // Transmitters REQUIRE valid AX.25 address
+                  aif->tx_ok = xbool;
+		  if (xbool && aif->callsign) {
+		    if (!validate_callsign_input(aif->callsign,xbool)) {  // Transmitters REQUIRE valid AX.25 address
 		      printf("%s:%d: ERROR: TX-OK 'TRUE' -- BUT PREVIOUSLY SET CALLSIGN IS NOT VALID AX.25 ADDRESS \n",
 			     cf->name, cf->linenum);
 		      continue;
@@ -774,24 +774,24 @@ int interface_config(struct configfile *cf)
 		  }
 
 		} else if (strcmp(name, "telem-to-is") == 0) {
-                  int bool;
-		  if (!config_parse_boolean(param1, &bool)) {
+                  int xbool;
+		  if (!config_parse_boolean(param1, &xbool)) {
 		    printf("%s:%d ERROR: Bad TELEM-TO-IS parameter value -- not a recognized boolean: %s\n",
 			   cf->name, cf->linenum, param1);
 		    have_fault = 1;
 		    break;
 		  }
-                  aif->telemeter_to_is = bool;
+                  aif->telemeter_to_is = xbool;
 
 		} else if (strcmp(name, "telem-to-rf") == 0) {
-                  int bool;
-		  if (!config_parse_boolean(param1, &bool)) {
+                  int xbool;
+		  if (!config_parse_boolean(param1, &xbool)) {
 		    printf("%s:%d ERROR: Bad TELEM-TO-RF parameter value -- not a recognized boolean: %s\n",
 			   cf->name, cf->linenum, param1);
 		    have_fault = 1;
 		    break;
 		  }
-                  aif->telemeter_to_rf = bool;
+                  aif->telemeter_to_rf = xbool;
 
 		} else if (strcmp(name,"timeout") == 0) {
 		  if (config_parse_interval(param1, &(aif->timeout) ) ||
-- 
2.50.1

